File: app/src/main/java/com/python/companion/db/typeconverters/DurationConverter.java
package com.python.companion.db.typeconverters;

import androidx.room.TypeConverter;

import java.time.Duration;
import java.time.temporal.ChronoUnit;

public class DurationConverter {
    @TypeConverter
    public static Duration durationFromDays(Long value) {
        return value == null ? null : Duration.of(value, ChronoUnit.DAYS);
    }

    @TypeConverter
    public static Long durationToDays(Duration duration) {
        return duration.toDays();
    }
}

File: db/src/main/java/org/python/db/typeconverters/DurationConverter.kt
package org.python.db.typeconverters

import androidx.room.TypeConverter
import java.time.Duration
import java.time.temporal.ChronoUnit


object DurationConverter {
    @TypeConverter
    fun durationFromDays(value: Long?): Duration? {
        return if (value == null) null else Duration.of(value, ChronoUnit.DAYS)
    }

    @TypeConverter
    fun durationToDays(duration: Duration): Long {
        return duration.toDays()
    }
}